wgpu-core/src/device/queue.rs

```diff
@@ -163,6 +166,13 @@ impl Drop for Queue {
             Ok(true) => {}
             // Note: If we don't panic here we are in UB land (destroying resources while they are still in use by the GPU).
             Ok(false) => {
+                // It's fine that we timed out on WebGL; GL objects can be deleted early as they
+                // will be kept around by the driver if GPU work hasn't finished.
+                // Moreover, the way we emulate read mappings on WebGL allows us to execute map_buffer earlier than on other
+                // backends since getBufferSubData is synchronous with respect to the other previously enqueued GL commands.
+                // TODO: Relying on this behavior breaks the clean abstraction wgpu-hal tries to maintain and
+                // we should find ways to improve this.
```

Comment:
todo: I don't think we should leave a `TODO` here on the final merge; we can just delete these lines after we file an issue.

Reply:
Filed: https://github.com/gfx-rs/wgpu/issues/6538